### PR TITLE
Update to 2.2.1. JB#57925

### PR DESCRIPTION
**Pull request by pvuorela**
About time we both get this updated and get rid of the git subtree setup.

Update in multiple steps:
1. Extracted the local changes to patches and made build use those. Patched upstream directory is identical to earlier directory with the modifications
2. Get rid of the extra copy of the sources
3. Move the tests.xml out of the patches and just do that in the .spec to avoid needing to port that to cmake
4. The actual update. Needed some extra adjustments and fixes. Some I could feed upstream.

Lot of noise because of the step 2. Could make sense to check the individual commits or the end result.

@mlehtima @spiiroin @llewelld

### CODE REVIEW

File: rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch [hunk 1]
```diff
@@ -1,19 +1,61 @@
...
+-    find_package(UDev REQUIRED)
+-    list(APPEND PLUGINS_LIBRARIES UDev::UDev)
++    # TODO: separate mce and udev
++    #find_package(UDev REQUIRED)
++    #list(APPEND PLUGINS_LIBRARIES UDev::UDev)
+ else()
+     list(APPEND PLUGINS_SOURCES src/mimhwkeyboardtracker_stub.cpp)
+ endif()
+ 
++
```
Comment (mlehtima) [inline rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch]: Nitpick, useless new line change.

File: rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch [hunk 2]
```diff
@@ -88,44 +137,26 @@ index b7126b5..9cb326d 100644
...
-     // Parse the evdev event and look for SW_TABLET_MODE status.
- 
-     struct input_event ev;
-@@ -130,14 +150,12 @@ void MImHwKeyboardTrackerPrivate::evdevEvent()
+@@ -128,11 +151,12 @@ void MImHwKeyboardTrackerPrivate::evdevEvent()
          evdevTabletModePending = -1;
          Q_EMIT stateChanged();
      }
--#endif
+-
```
Comment (mlehtima) [inline rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch]: Nitpick: useless new line change left from removing endif.

File: rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch [hunk 3]
```diff
@@ -88,44 +137,26 @@ index b7126b5..9cb326d 100644
...
  }
  
  void MImHwKeyboardTrackerPrivate::tryEvdevDevice(const char *device)
  {
-     Q_UNUSED(device);
- 
--#ifndef HAVE_CONTEXTSUBSCRIBER
++    Q_UNUSED(device);
```
Comment (mlehtima) [inline rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch]: Is this change needed? It is within an ifdef which I assume disables this whole function in our case. A bit difficult to see in the diff but final patch shows better.

File: rpm/0002-enable-systemd-activation.patch [hunk 4]
```diff
@@ -11,13 +11,13 @@ apply patch enable-systemd-activation.patch
...
 diff --git a/connection/org.maliit.server.service.in b/connection/org.maliit.server.service.in
-index 6b38706..0726771 100644
+index 6a3e921..f387838 100644
 --- a/connection/org.maliit.server.service.in
 +++ b/connection/org.maliit.server.service.in
 @@ -1,4 +1,5 @@
  [D-BUS Service]
  Name=org.maliit.server
- Exec=@BINDIR@/maliit-server @MALIIT_SERVER_ARGUMENTS@
+ Exec=@CMAKE_INSTALL_PREFIX@/bin/maliit-server @MALIIT_SERVER_ARGUMENTS@
```
Comment (spiiroin) [inline rpm/0002-enable-systemd-activation.patch:20]: Since systemd activation is used, this ought to be Exec=/usr/bin/false
... or is it already done in another patch?

**Review by spiiroin — APPROVED** [review]
Did not spot anything obvious. Builds, installs and seems to work ok too.

File: rpm/0002-enable-systemd-activation.patch (hunk 4 above)
Comment (Thaodan) [reply rpm/0002-enable-systemd-activation.patch:20]: Isn't using SystemdService= enough?

File: rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch (hunk 3 above)
Comment (pvuorela) [reply rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch]: There was three commits touching hw keyboard tracking. Couple two doing first small adjustments to context subscriber path and the last one for switching to mce. Now I just merged these together and didn't bother removing unnecessary content as there didn't appear to be too much of it.

Indeed this could be skipped. Maybe I'll do that if I end up updating the PR for some other reason.

File: rpm/0002-enable-systemd-activation.patch (hunk 4 above)
Comment (pvuorela) [reply rpm/0002-enable-systemd-activation.patch:20]: It's switched to false in later patch, as it was earlier in the git commits.

Comment (spiiroin) [reply rpm/0002-enable-systemd-activation.patch:20]: > Isn't using SystemdService= enough?

While on happy path, yes. Using a dummy Exec just ensures that there is no way dbus daemon can autolaunch a service without going through systemd.

File: rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch (hunk 3 above)
Comment (pvuorela) [reply rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch]: Done.
